src/mod/asr_tts/mod_unimrcp/mod_unimrcp.c

```diff
@@ -1926,7 +1930,7 @@ static apt_bool_t speech_on_channel_add(mrcp_application_t *application, mrcp_se
 	} else {
 		descriptor = mrcp_application_source_descriptor_get(channel);
 	}
-	if (!descriptor) {
+	if (!descriptor || schannel->channel_destroyed) {
```

Comment:
No need to check schannel->channel_destroyed here since we already validated it above